homeassistant/components/shelly/__init__.py

```diff
@@ -82,19 +90,29 @@ def __init__(self, hass, entry, device: aioshelly.Device):
             hass,
             _LOGGER,
             name=device.settings["name"] or device.settings["device"]["hostname"],
-            update_interval=timedelta(seconds=5),
         )
         self.hass = hass
         self.entry = entry
         self.device = device
+        self._update_from_multicast = False
+
+        self.device.subscribe_updates(self._multicast_update)
+
+    def _multicast_update(self, data):
```

Comment:
To indicate that it's called from async context

```suggestion
    @callback
    def _async_multicast_update(self, data):
```